mruby fails tests on master

**Describe the bug**
`mruby` cannot be built from current master due to failing tests.

**To Reproduce**
1. `nix-build '<nixpkgs>' -A mruby`

**Additional context**
Relevant logs:
```
>>> Bintest test <<<
bintest - Command Binary Test

.......FFF......FF..........................
Fail: regression for #1564 (mrbgems: mruby-bin-mruby)
 - Assertion[1]
    Fail: assert_mruby (mrbgems: mruby-bin-mruby)
     - Assertion[1-2] standard error.
        Expected /\A-e:1:2: syntax error, .*\n\z/ to be === "-e:1:2: syntax error\n".
 - Assertion[2]
    Fail: assert_mruby (mrbgems: mruby-bin-mruby)
     - Assertion[2-2] standard error.
        Expected /\A-e:1:3: syntax error, .*\n\z/ to be === "-e:1:3: syntax error\n".
Fail: mruby -c option (mrbgems: mruby-bin-mruby)
 - Assertion[2]
    Fail: assert_mruby (mrbgems: mruby-bin-mruby)
     - Assertion[2-2] standard error.
        Expected /\A-e:1:7: syntax error, .*\n\z/ to be === "-e:1:7: syntax error\n".
  Total: 44
     OK: 39
     KO: 5
  Crash: 0
Warning: 0
   Skip: 0
   Time: 0.16 seconds
rake aborted!
Command Failed: [ruby test/bintest.rb /build/source/mrbgems/mruby-compiler /build/source/mrbgems/mruby-bin-mirb /build/source/mrbgems/mruby-bin-mruby /build/source/mrbgems/mruby-bin-strip]

make: *** [Makefile:12: test] Error 1
```

**Notify maintainers**
@nicknovitski 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.118, NixOS, 20.09.git.452fe6ac8e8M (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.6`
 - channels(root): `"nixos-20.09pre218613.ae6bdcc5358"`
 - channels(noneucat): `""`
 - nixpkgs: `/home/noneucat/Documents/NixPkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: mruby
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/compilers/mruby/bison-36-compat.patch
+++ b/pkgs/development/compilers/mruby/bison-36-compat.patch
@@ -1,0 +1,59 @@
+From acab088fd6af0b2ef2df1396aeb93bfc2e020fa5 Mon Sep 17 00:00:00 2001
+From: "Yukihiro \"Matz\" Matsumoto" <matz@ruby.or.jp>
+Date: Mon, 27 Apr 2020 18:52:43 +0900
+Subject: [PATCH 1/2] Updating `parse.y for recent `bison` (retry).
+
+---
+ mrbgems/mruby-compiler/core/parse.y | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/mrbgems/mruby-compiler/core/parse.y b/mrbgems/mruby-compiler/core/parse.y
+index 6a1faf4e..2a4f740e 100644
+--- a/mrbgems/mruby-compiler/core/parse.y
++++ b/mrbgems/mruby-compiler/core/parse.y
+@@ -1323,7 +1323,7 @@ heredoc_end(parser_state *p)
+ 
+ %}
+ 
+-%pure-parser
++%define api.pure
+ %parse-param {parser_state *p}
+ %lex-param {parser_state *p}
+ 
+-- 
+2.27.0
+
+From 3cc682d943b29e84928a847a23f411ddbace74b7 Mon Sep 17 00:00:00 2001
+From: "Yukihiro \"Matz\" Matsumoto" <matz@ruby.or.jp>
+Date: Fri, 15 May 2020 12:30:13 +0900
+Subject: [PATCH 2/2] Remove `YYERROR_VERBOSE` which no longer supported since
+ `bison 3.6`.
+
+Instead we added `%define parse.error verbose`.
+---
+ mrbgems/mruby-compiler/core/parse.y | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/mrbgems/mruby-compiler/core/parse.y b/mrbgems/mruby-compiler/core/parse.y
+index 2a4f740e..eee6a5e5 100644
+--- a/mrbgems/mruby-compiler/core/parse.y
++++ b/mrbgems/mruby-compiler/core/parse.y
+@@ -9,7 +9,6 @@
+ #ifdef PARSER_DEBUG
+ # define YYDEBUG 1
+ #endif
+-#define YYERROR_VERBOSE 1
+ #define YYSTACK_USE_ALLOCA 1
+ 
+ #include <ctype.h>
+@@ -1323,6 +1322,7 @@ heredoc_end(parser_state *p)
+ 
+ %}
+ 
++%define parse.error verbose
+ %define api.pure
+ %parse-param {parser_state *p}
+ %lex-param {parser_state *p}
+-- 
+2.27.0
+

--- a/pkgs/development/compilers/mruby/default.nix
+++ b/pkgs/development/compilers/mruby/default.nix
@@ -1,17 +1,17 @@
-{ stdenv, ruby, bison, fetchFromGitHub }:
+{ stdenv, ruby, bison, rake, fetchFromGitHub }:
 
 stdenv.mkDerivation rec {
   pname = "mruby";
-  version = "2.1.0";
+  version = "2.1.1";
 
   src = fetchFromGitHub {
     owner   = "mruby";
     repo    = "mruby";
     rev     = version;
-    sha256  = "1y072c7dh9jf8xwy7kia6cb4dkpspq4zf24ssn7zm5f46p4waxni";
+    sha256  = "gEEb0Vn/G+dNgeY6r0VP8bMSPrEOf5s+0GoOcnIPtEU=";
   };
 
-  nativeBuildInputs = [ ruby bison ];
+  nativeBuildInputs = [ ruby bison rake ];
 
   patches = [ ./bison-36-compat.patch ];
 

--- a/pkgs/development/compilers/mruby/default.nix
+++ b/pkgs/development/compilers/mruby/default.nix
@@ -13,6 +13,8 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ ruby bison ];
 
+  patches = [ ./bison-36-compat.patch ];
+
   # Necessary so it uses `gcc` instead of `ld` for linking.
   # https://github.com/mruby/mruby/blob/35be8b252495d92ca811d76996f03c470ee33380/tasks/toolchains/gcc.rake#L25
   preBuild = if stdenv.isLinux then "unset LD" else null;

